### PR TITLE
bug(AUTH-CPC-391): handle errors for GET

### PR DESCRIPTION
**Pull request by Trilikin21**
Jira Code : https://champlainsaintlambert.atlassian.net/jira/software/c/projects/CPC/boards/16/backlog?view=detail&selectedIssue=CPC-390&issueLimit=100

Implemented error handling for the getByUserId method and its errors

### CODE REVIEW

File: auth-service/src/main/java/com/petclinic/auth/Config/SpringSecurityConfig.java [hunk 1]
```diff
@@ -73,6 +73,7 @@ protected void configure(HttpSecurity http) throws Exception {
                 .authorizeRequests().antMatchers("/roles").hasRole("ADMIN")
                 .antMatchers(HttpMethod.POST,"/users").permitAll()
                 .antMatchers(HttpMethod.GET,"/users/*").permitAll()
+                .antMatchers(HttpMethod.GET,"/users").permitAll()
                 .antMatchers(HttpMethod.DELETE,"/users/*").permitAll()
```
Comment (Fube) [inline auth-service/src/main/java/com/petclinic/auth/Config/SpringSecurityConfig.java:77]: Why permit access to everyone on these endpoints?
Also, if you're going to bind POST, GET, and DELETE you might as well simplify it to
```suggestion
                .antMatchers("/users").permitAll()
```

File: auth-service/src/main/java/com/petclinic/auth/User/UserServiceImpl.java [hunk 2]
```diff
@@ -59,10 +61,15 @@
 
     @Override
     public User getUserById(long id) {
-        User entity  = userRepo.findById(id)
-                .orElseThrow(() -> new NotFoundException("No user found for userID" + id));
-        log.info("User getUserById: found userId: {}", entity.getId());
-        return entity;
+        if (id <= 0){
+            throw new InvalidInputException("Id cannot be a negative number for " + id);
```
Comment (Fube) [inline auth-service/src/main/java/com/petclinic/auth/User/UserServiceImpl.java:65]: Some databases might allow negative numbers for ID

File: auth-service/src/test/java/com/petclinic/auth/User/AuthServiceUserControllerTests.java [hunk 3]
```diff
@@ -412,4 +415,44 @@ void bad_login_throw_incorrect_password_exception() throws Exception {
...
+        long id = 23212;
+        NotFoundException notFoundException = new NotFoundException("No user found for userID " + id);
+        given(userService.getUserById(id)).willThrow(notFoundException);
+        mockMvc.perform(get("/users/" + id))
+                .andExpect(status().isNotFound())
+                .andExpect(jsonPath("$.statusCode").value(404))
+                .andExpect(jsonPath("$.message").value("No user found for userID " + id));
+
+
+    }
```
Comment (Fube) [inline auth-service/src/test/java/com/petclinic/auth/User/AuthServiceUserControllerTests.java:457]: Beautiful

**Review by Fube — DISMISSED** [review]

File: auth-service/src/main/java/com/petclinic/auth/Config/SpringSecurityConfig.java (hunk 1 above)
Comment (Trilikin21) [reply auth-service/src/main/java/com/petclinic/auth/Config/SpringSecurityConfig.java]: this has been resolve. As for the security settings, I am allowing anybody to use the API for testing for now.

**Review by JordanAlbayrak — APPROVED** [review]
Looks gucci. I'm worried about some conflicts with Nariman's PR since you guys modified similar files but we'll see when the time comes.

File: auth-service/src/main/java/com/petclinic/auth/Config/GlobalControllerExceptionHandlerConfig.java [hunk 4]
```diff
@@ -2,20 +2,37 @@
...
     public HTTPErrorMessage resourceNotFoundException(IncorrectPasswordException ex, WebRequest request) {
 
         return new HTTPErrorMessage(401, ex.getMessage());
     }
+
+    @ExceptionHandler(value = NotFoundException.class)
+    @ResponseStatus(value = NOT_FOUND)
+    public HTTPErrorMessage NotFoundException(NotFoundException ex, WebRequest request) {
+
+        return new HTTPErrorMessage(404, ex.getMessage());
```
Comment (Fube) [inline auth-service/src/main/java/com/petclinic/auth/Config/GlobalControllerExceptionHandlerConfig.java:29]: Could use HttpStatus enum for better readability
Check out 
https://github.com/cgerard321/champlain_petclinic/blob/e91d80b9b4689baf6e97dd8374276681178383e8/auth-service/src/main/java/com/petclinic/auth/Config/GlobalControllerExceptionHandlerConfig.java#L28

**Review by Fube — APPROVED** [review]
LGTM 🤙

**Pull request merged**